From 515c1e4d3eaa7c6bacfa9908d39bfad7b507f244 Mon Sep 17 00:00:00 2001
From: "Langer, Thomas" <tlanger@maxlinear.com>
Date: Thu, 16 Mar 2023 19:59:28 +0530
Subject: [PATCH 156/500] OpenWrt:
 724-net-phy-aquantia-Add-AQR113-driver-support.patch

PONRTSYS-8842: aquantia: Add AQR113 driver support

Add a new entry for AQR113 PHY_ID

Signed-off-by: Langer, Thomas <tlanger@maxlinear.com>

Change-Id: Ic1e285878d25ddb3c278697c6d9e35d79bfe9433
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
(cherry picked from commit 13a3dff9b35bc19693b18a47962a419c2a30aacc)
---
 drivers/net/phy/aquantia_main.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/net/phy/aquantia_main.c b/drivers/net/phy/aquantia_main.c
index 50a813b72b7a..c9061676155e 100644
--- a/drivers/net/phy/aquantia_main.c
+++ b/drivers/net/phy/aquantia_main.c
@@ -28,6 +28,7 @@
 #define PHY_ID_AQR813   0x31c31cb2
 #define PHY_ID_AQR112	0x03a1b662
 #define PHY_ID_AQR412	0x03a1b712
+#define PHY_ID_AQR113	0x31c31c40
 
 #define MDIO_PHYXS_VEND_IF_STATUS		0xe812
 #define MDIO_PHYXS_VEND_IF_STATUS_TYPE_MASK	GENMASK(7, 3)
@@ -1019,6 +1020,14 @@ static struct phy_driver aqr_driver[] = {
 	.get_strings	= aqr107_get_strings,
 	.get_stats	= aqr107_get_stats,
 },
+{
+	PHY_ID_MATCH_MODEL(PHY_ID_AQR113),
+	.name		= "Aquantia AQR113",
+	.config_aneg	= aqr_config_aneg,
+	.config_intr	= aqr_config_intr,
+	.handle_interrupt = aqr_handle_interrupt,
+	.read_status	= aqr107_read_status,
+},
 };
 
 module_phy_driver(aqr_driver);
@@ -1035,6 +1044,7 @@ static struct mdio_device_id __maybe_unused aqr_tbl[] = {
 	{ PHY_ID_MATCH_MODEL(PHY_ID_AQR412) },
 	{ PHY_ID_MATCH_MODEL(PHY_ID_AQR113C) },
 	{ PHY_ID_MATCH_MODEL(PHY_ID_AQR813) },
+	{ PHY_ID_MATCH_MODEL(PHY_ID_AQR113) },
 	{ }
 };
 
-- 
2.34.1

